src/app/public/src/modules/shared/route-metadata.service.ts

```diff
@@ -18,12 +18,17 @@ export class StacheRouteMetadataService {
   constructor(
     @Inject(STACHE_ROUTE_METADATA_SERVICE_CONFIG)
     public metadata: any[]) {
-      this.metadata.forEach(route => {
-        if (route.order) {
-          route.order = numberConverter(route.order);
-        }
-      });
+      this.metadata.forEach((route: any) => this.validateNavOrder(route));
     }
+
+  private validateNavOrder(route: any): void {
+    if (route.hasOwnProperty('order')) {
+      route.order = numberConverter(route.order);
+      if (route.order <= 0 || route.order !== parseInt(route.order, 10)) {
```

Comment:
I thought about this, and decided against supporting 0.  It created the same problem supporting `-` numbers did.   If I had a `navOrder='0'` somewhere in a large list.   Later I wanted to move a nav item around to the 4th position in a list,   I would immediately try `navOrder='4'`... and then be surprised when it went to the 5th position.     

By not supporting `0`, this allows our `navOrder` to directly reflect it's position in the list.   (unless there are duplicate navOrder items).